components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlinx/coroutines/flow/Flow.kt

```diff
@@ -47,6 +50,34 @@ fun <T, R> Flow<T>.ifChanged(transform: (T) -> R): Flow<T> {
     }
 }
 
+/**
+ * Returns a [Flow] containing only changed elements of the lists of the original [Flow].
+ *
+ * Example:
+ * ```
+ * Block: x -> x[0]  // Map to first character of input
+ * Original Flow: list("banana"), list("bandanna", "bus"), list("apple", "big", "coconut", "circle"), list("home")
+ * Mapped: ("b"), ("b", "b"), ("a", "b", "c", "c"), ("h")
+ * Returned Flow: "banana", "bus", "apple", "coconut", "circle", "home"
+ * ```
+ */
+@FlowPreview
+fun <T, R> Flow<List<T>>.filterChanged(transform: (T) -> R): Flow<T> {
```

Comment:
Interesting, you flatten internally and go directly from `Flow<List<T>>` to `Flow<T>`. Originally I was expecting this to return the filtered `Flow<List<T>>` and then the app/component code can decide whether to use the list or flatten. However right now I am not sure if that will be a valid use case. Anyhow, I guess we can always adapt later.

Reply:
Yes, I had started with that, but this made the consuming code easier. Maybe it's too "magical" and we should add another step to the API to flatten. Probably worth revisiting once we have more use cases...